libafl_frida/src/asan/asan_rt.rs

```diff
@@ -483,40 +473,36 @@ impl AsanRuntime {
     #[allow(clippy::too_many_lines)]
     pub fn register_hooks(&mut self, gum: &Gum) {
         let mut interceptor = Interceptor::obtain(gum);
-
         macro_rules! hook_func {
             ($lib:expr, $name:ident, ($($param:ident : $param_type:ty),*), $return_type:ty) => {
                 paste::paste! {
                     log::trace!("Hooking {}", stringify!($name));
 
                     let target_function = frida_gum::Module::find_export_by_name($lib, stringify!($name)).expect("Failed to find function");
-                    self.hooks.insert(stringify!($name).to_string(), target_function);
+
+                    extern "system" { fn $name($($param: $param_type),*) -> $return_type; }
```

Comment:
It looks like  you have moved *back* to the previous implementation of `hook_func`. There is a reason that I made the changes that I did:

1. Declaring the original functions as `extern "system"` directly requires them to be linked at link time. This would require us to link with a whole bunch of libraries which we can't/don't want to do. The new solution is to store the original function pointers in a HashMap and retrieve and transmute them at *run time*.
2. It is necessary to hook *all* calls to functions while the `hooks_enabled` variable is set. Not just from stalked code (your revert at line 489 below). The suppression should probably be enabled, but otherwise the new implementation is preferred.

In general, we need to go back to the new implementation, and if there are issues, fix them. What are the issues you encountered?